### DISCUSSION TITLE
F-Droid inclusion

### DISCUSSION THREAD

**Discussion by momobobe** [fcitx5-android/fcitx5-android#114] [General | 讨论]
Hello,

Since this is an open-source app, would you approve to have it included to [F-Droid](https://f-droid.org/)? It's a strong community (actually almost the unique) focusing on distributing open-source Android apps, and you can find some other CJK IMEs like Trime, Guileless, Mozc, etc. already listed there.

You can find documentation [here](https://f-droid.org/docs/).

If you're interested, feel free to contact @ linsui (who speaks Chinese) or @ Poussinou for further details.

Thanks for your attention and it wouldn't take too much time since their team would help you.

PS: Apps on F-Droid are signed by F-Droid by default, but you can ask for the case of reproducible build, then they just verify if your builds are reproducible and simply distribute the builds signed by you, in that way there would be no problem of distributing packages with different signatures.

PPS: just a suggestion, no matter if you agree to add the app to F-Droid, please regularly tag your releases so that others can easily know if there is any new version. Many thanks

**Comment by rocka** [comment]
Thanks for the suggestion, and we are happy to have the App included to F-Droid. But considering the project is still in early alpha stage, there aren't any release schedule, the build process is subject to change as well. So we think it maybe too early to be included in F-Droid.

The App heavily relies on higher version of Android NDK (namely >= r23b), and we have several prebuilt static libraires (https://github.com/fcitx5-android/prebuilt). Is it suitable for F-Droid build?

**Comment by momobobe** [comment]
Okay, I get it. As for your question, if they can build your prebuilt libraries, that should be fine (also maybe you can adjust the build process to merge them into one project, but I'm not a developer so not sure).

When you feel it suitable to release on F-Droid, you can ask github.com/linsui for assistance. (I'm not responsible for later process, so, yeah, my job is done)

Thanks again for your attention.

**Comment by linsui** [comment]
Wow, I was not aware that there is another FOSS Chinese IME for Android besides Trime. This looks great! I'm very willing to help package it for F-Droid. Currently there is only one general Chinese IME on F-Droid. So even if this one is not very mature it's much better than nothing.

**Comment by linsui** [comment]
Since you have published the initial release, is this good to be published on F-Droid?

**Reply by rocka** [reply]
I don't think it's ready yet. Several bugs were fixed after tagging https://github.com/fcitx5-android/fcitx5-android/releases/tag/0.0.3 , also some quality-of-life features were included. But more importantly, the build process hasn't been changed, we still rely on https://github.com/fcitx5-android/prebuilt to build native libraries.

**Reply by linsui** [reply]
> Several bugs were fixed after tagging https://github.com/fcitx5-android/fcitx5-android/releases/tag/0.0.3 , also some quality-of-life features were included.

Software always has bug. :shrug: We can add some notes in the description. It won't break user's system anyway.

> But more importantly, the build process hasn't been changed, we still rely on https://github.com/fcitx5-android/prebuilt to build native libraries.

We can build them with https://github.com/fcitx5-android/prebuilder before building the app.

**Reply by berberman** [reply]
We are happy to shelve the app on F-Droid official repository, but it's not clear how to build it with F-Droid's regulation. Do you have any idea?

**Reply by linsui** [reply]
I'll have a try.

**Reply by linsui** [reply]
https://gitlab.com/fdroid/fdroiddata/-/merge_requests/12373 I drafted an MR.

**Comment by linsui** [comment]
It will be published in 1 week.
